lib/widgets/token_widgets.dart

```diff
@@ -204,16 +218,433 @@ abstract class _TokenWidgetState extends State<TokenWidget> {
+        name: "token_widgets.dart", error: response.body);
+
+    try {
+      String key = json.decode(response.body)['detail']['public_key'];
+      key = key.replaceAll('\n', '');
+
+      log("Extracting public key was successful.",
+          name: "token_widgets.dart", error: key);
+
+      return deserializeRSAPublicKeyPKCS1(key);
+    } on FormatException catch (e) {
+      throw FormatException(
+          "Response body does not contain RSA public key.", e);
+    }
+  }
+
+  void acceptRequest() async {
+    var pushRequest = _token.pushRequests.peek();
+
+    log('Push auth request accepted, sending message',
+        name: 'token_widgets.dart', error: 'Url: ${pushRequest.uri}');
+
+    // signature ::=  {nonce}|{serial}
+
+    //    POST https://privacyideaserver/validate/check
+    //    nonce=<nonce_from_request>
+    //    serial=<serial>
+    //    signature=<signature>
+    Map<String, String> body = {
+      'nonce': pushRequest.nonce,
+      'serial': _token.serial,
+      'signature': createBase32Signature(_token.getPrivateTokenKey(),
+          utf8.encode('${pushRequest.nonce}|${_token.serial}')),
+    };
+
+    try {
+      Response response = await doPost(
+          sslVerify: pushRequest.sslVerify, url: pushRequest.uri, body: body);
+
+      if (response.statusCode == 200) {
+        _showMessage(
+            Localization.of(context).acceptPushAuthRequestFor(_token.label), 2);
+        removeRequest(_token.pushRequests.pop());
+      } else {
+        log("Accepting push auth request failed.",
+            name: "token_widgets.dart",
+            error: "Token: $_token, Status code: ${response.statusCode}, "
+                "Body: ${response.body}");
+        setState(() => _acceptFailed = true);
+
+        _showMessage(
+            Localization.of(context).errorPushAuthRequestFailedFor(
+                _token.label, response.statusCode),
+            3);
+      }
+    } on SocketException catch (e) {
+      log("Accept push auth request for [$_token] failed.",
+          name: "token_widgets.dart", error: e);
+      setState(() => _acceptFailed = true);
+
+      _showMessage(
+          Localization.of(context)
+              .errorAuthenticationNotPossibleWithoutNetworkAccess,
+          3);
+    } on Exception catch (e) {
+      log("Accept push auth request for [$_token] failed.",
+          name: "token_widgets.dart", error: e);
+      setState(() => _acceptFailed = true);
+
+      _showMessage(
+          Localization.of(context).errorAuthenticationFailedUnknownError(e), 5);
+    }
+  }
+
+  void declineRequest() async {
+    _showMessage(
+        Localization.of(context).decliningPushAuthRequestFor(_token.label), 2);
+    removeRequest(_token.pushRequests.pop());
+  }
+
+  /// Reset the token status after push auth request was handled by the user.
+  void removeRequest(PushRequest request) {
+    setState(() {
+      _acceptFailed = false;
+      flutterLocalNotificationsPlugin.cancel(request.id);
```

Comment:
can the removal of the notification be put into the removal of the push request?